.github/workflows/ci.yml

```diff
@@ -0,0 +1,58 @@
+name: AppConfiguration-DotnetProvider CI
+
+on:
+  push:
+    branches:
+      - main
+      - preview
+      - release/*
+  pull_request:
+    branches:
+      - main
+      - preview
+      - release/*
+
+permissions:
+  security-events: write
```

Comment:
what's this for?

Reply:
From what I understand, it's to allow codeql to push any issues to [this alerts page](https://github.com/Azure/AppConfiguration-DotnetProvider/pull/582/checks?check_run_id=27765974502). I thought we might want to run codeql since it was on our other pipelines and it's supported by github.

Reply:
> Write permissions might also be necessary if you want to automatically create or update security advisories based on the analysis results.

It makes sense to have write permission.